palisade-service/src/main/java/uk/gov/gchq/palisade/service/palisade/service/SimplePalisadeService.java

```diff
@@ -76,62 +77,120 @@ public SimplePalisadeService(final AuditService auditService, final UserService
         this.aggregationService = resultAggregationService;
     }
 
-    @Override
-    public CompletableFuture<DataRequestResponse> registerDataRequest(final RegisterDataRequest request) {
-        requireNonNull(request, "request");
-        final RequestId originalRequestId = request.getId();
-        LOGGER.debug("Registering data request: {}, {}", request, originalRequestId);
-
-        AuditRequest.RegisterRequestExceptionAuditRequest.IException auditException = AuditRequest.RegisterRequestExceptionAuditRequest
-                .create(originalRequestId)
+    private AuditRequest.RegisterRequestExceptionAuditRequest.IException getAuditException(final RegisterDataRequest request) {
+        return AuditRequest.RegisterRequestExceptionAuditRequest
+                .create(request.getId())
                 .withUserId(request.getUserId())
                 .withResourceId(request.getResourceId())
                 .withContext(request.getContext());
+    }
 
+    private CompletableFuture<User> getFutureUser(final RegisterDataRequest request) {
         final GetUserRequest userRequest = new GetUserRequest().userId(request.getUserId());
-        userRequest.setOriginalRequestId(originalRequestId);
-        LOGGER.debug("Getting user from userService: {}", userRequest);
-        final CompletableFuture<User> user = userService.getUser(userRequest)
+        userRequest.setOriginalRequestId(request.getId());
+        return userService
+                .getUser(userRequest)
                 .exceptionally(ex -> {
-                    LOGGER.error("Auditing exception from UserService", ex);
-                    auditService.audit(auditException.withException(ex).withServiceClass(UserService.class));
-                    throw new RuntimeException(ex);
+                    LOGGER.error("Auditing exception from userService", ex);
+                    auditService.audit(getAuditException(request)
+                            .withException(ex)
+                            .withServiceClass(UserService.class));
+                    throw new RegisterRequestException("Exception from userService", ex);
```

Comment:
Again logging error, logging audit and re-throwing. Suggest removing logging or maybe just log the error message rather than whole exception

Reply:
same applies in many places in this file